python/ray/tests/test_multiprocessing.py

```diff
@@ -562,6 +563,20 @@ def f(args):
     pool.join()
 
 
+def test_deadlock_avoidance_in_recursive_tasks():
+    def poolit_a(_):
+        with Pool(ray_address="auto") as pool:
+            return list(pool.map(math.sqrt, range(0, 2, 1)))
+
+    def poolit_b():
+        with Pool(ray_address="auto") as pool:
+            return list(pool.map(poolit_a, range(2, 4, 1)))
+
+    ray.init(num_cpus=1)
```

Comment:
nit: can we initialize ray with a fixture if possible? Ideally we can use `ray_start_shared_local_modes`?